.github/workflows/release.yml

```diff
@@ -32,18 +32,70 @@ jobs:
         run: cd main/ && GOOS=linux GOARCH=amd64 go build -o ~/main/bin/amd64-linux-openshiftplugin-${{ github.event.inputs.version }} .
 
       - name: Build openshift plugin darwin amd64
-          env:
-            GOPROXY: "https://proxy.golang.org"
-          run: cd main/ && GOOS=linux GOARCH=amd64 go build -o ~/main/bin/amd64-linux-openshiftplugin-${{ github.event.inputs.version }} .
+        env:
+          GOPROXY: "https://proxy.golang.org"
+        run: cd main/ && GOOS=darwin GOARCH=amd64 go build -o ~/main/bin/amd64-darwin-openshiftplugin-${{ github.event.inputs.version }} .
 
       - name: Build openshift plugin darwin arm64
-          env:
-            GOPROXY: "https://proxy.golang.org"
-          run: cd main/ && GOOS=linux GOARCH=amd64 go build -o ~/main/bin/amd64-linux-openshiftplugin-${{ github.event.inputs.version }} .
+        env:
+          GOPROXY: "https://proxy.golang.org"
+        run: cd main/ && GOOS=darwin GOARCH=arm64 go build -o ~/main/bin/arm64-darwin-openshiftplugin-${{ github.event.inputs.version }} .
 
       - name: release
         uses: ncipollo/release-action@v1
         with:
           artifacts: "~/main/bin/*"
           token: ${{ secrets.GITHUB_TOKEN }}
-          tag: ${{ github.event.inputs.version }}
\ No newline at end of file
+          tag: ${{ github.event.inputs.version }}
+
+      - name: Checkout crane-plugins
+        uses: actions/checkout@v2
+        with:
+          repository: ${{ github.repository_owner }}/crane-plugins
+          token: ${{ secrets.PLUGIN_RELEASE }}
+
+      - name: Updating index file and adding manifest
+        run: |
+          cat << EOF >> index.yml
+          OpenshiftPlugin-${{ github.event.inputs.version }}: https://github.com/${{ github.repository_owner }}/crane-plugins/raw/main/plugins/OpenshiftPlugin-${{ github.event.inputs.version }}.yml
+          EOF
+          cat << EOF >> plugins/OpenshiftPlugin-${{ github.event.inputs.version }}.yml
+          name: OpenshiftPlugin
+          shortDescription: OpenshiftPlugin
+          description: this is OpenshiftPlugin
+          version: ${{ github.event.inputs.version }}
+          binaries:
+            - os: linux
+              arch: amd64
+              uri: https://github.com/${{ github.repository }}/releases/download/${{ github.event.inputs.version }}/amd64-linux-openshiftplugin-${{ github.event.inputs.version }}
+            - os: darwin
+              arch: amd64
+              uri: https://github.com/${{ github.repository }}/releases/download/${{ github.event.inputs.version }}/amd64-darwin-openshiftplugin-${{ github.event.inputs.version }}
+            - os: darwin
+              arch: arm64
+              uri: https://github.com/${{ github.repository }}/releases/download/${{ github.event.inputs.version }}/arm64-darwin-openshiftplugin-${{ github.event.inputs.version }}
+          optionalFields:
```

Comment:
Probably want a jira, or follow-up task to generate this from source so we don't have to manually keep multiple files in sync.

The problem I'm imagining is one where you add a new optional field to the openshift plugin but don't add it here...from the plugin-manager's perspective does it even exist :sunglasses: .

Reply:
Agree!

Reply:
Yes -- as I mentioned before. This is fine *for now*, but reasonably shortly we need this automated. Probably there should be a workflow task to actually call the built-for-release plugin and get its own optional-fields output since this is what the crane tool will report for optional fields once it's installed.